### PR TITLE
Terminate if S3 buffer is not finalized

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by novikd — APPROVED** [review]

File: src/IO/WriteBufferFromS3.h [hunk 1]
```diff
@@ -105,6 +105,7 @@ class WriteBufferFromS3 final : public BufferWithOwnMemory<WriteBuffer>
     std::vector<String> part_tags;
 
     bool is_prefinalized = false;
+    std::atomic<bool> is_finalized = false;
```
Comment (antonio2368) [inline src/IO/WriteBufferFromS3.h:108]: why is it atomic? Isn't the WriteBufferFromS3 non thread safe?

Comment (alesapin) [reply src/IO/WriteBufferFromS3.h:108]: Agree

**Comment by alesapin** [comment]
@Mergifyio update

**Comment by alesapin** [comment]
Stress test timeouts fixed in master.

**Pull request merged**